hyperspy/drawing/markers.py

```diff
@@ -591,12 +591,11 @@ def update(self):
             self.collection.set(**kwds)
 
     def _initialize_collection(self):
-        if self.collection is None:
-            self.collection = self._collection_class(
-                **self.get_data_position(),
-                offset_transform=self.offset_transform,
+        self.collection = self._collection_class(
```

Comment:
In which situation is it necessary? Considering that the `collection` attribute is public, if an user change some properties of this object directly, creating a new one will break these, and currently this is happening when closing/opening a plot.

Reply:
@ericpre this is why the example for editing the collection was failing. 

Maybe this isn't the right thing to do but 


```python
import hyperspy.api as hs
import numpy as np

# Create a Signal2D with 2 navigation dimensions
rng = np.random.default_rng(0)
data = np.arange(15*100*100).reshape((15, 100, 100))
s = hs.signals.Signal2D(data)

#%%
# Create text marker

# Define the position of the texts
offsets = np.stack([np.arange(0, 100, 10)]*2).T + np.array([5,]*2)
texts = np.array(['a', 'b', 'c', 'd', 'e', 'f', 'g', 'f', 'h', 'i'])

m = hs.plot.markers.Texts(
    offsets=offsets,
    texts=texts,
    sizes=3,
    )

s.plot()
s.add_marker(m)

s.plot()
s.add_marker(m)

s.plot()
s.add_marker(m) # Not sure why this fails on the 3rd time?

```

fails on the third time because it tries to add the collection to the same plot twice.  It might be an issue with closing the plot or not correctly cleaning up.  Either way the `marker._initialize_collection()` function should only be called once when the `marker.plot()`function is called.

Reply:
This argues the collection should be private I think...

Reply:
Done! :)